bin-src/lib/comparePackageManifests.test.ts

```diff
@@ -0,0 +1,184 @@
+  expect(
+    arePackageDependenciesEqual(
+      { dependencies: { a: '1', b: '2', c: '3' } },
+      { dependencies: { c: '3', a: '1', b: '2' } }
+    )
+  ).toBe(true);
+});
+
+it('returns false if devDependencies are different', () => {
+  expect(
+    arePackageDependenciesEqual(
+      { devDependencies: { a: '1', b: '2' } },
+      { devDependencies: { a: '1', b: '2.2' } }
+    )
+  ).toBe(false);
+});
+
+it('returns false if peerDependencies are different', () => {
+  expect(
+    arePackageDependenciesEqual(
+      { peerDependencies: { a: '1', b: '2' } },
+      { peerDependencies: { a: '1', b: '2.2' } }
+    )
+  ).toBe(false);
+});
+
+it("returns true if differing object fields aren't dependency-related", () => {
+  expect(
+    arePackageDependenciesEqual(
+      {
+        scripts: {
+          test: 'yarn jest --watch',
+        },
+        dependencies: { a: '1' },
+      },
+      {
+        scripts: {
+          test: 'yarn jest --watch',
+          build: 'yarn esbuild',
+        },
+        dependencies: { a: '1' },
+      }
+    )
+  ).toBe(true);
+});
+
+it("returns true if differing non-object fields aren't dependency-related", () => {
+  expect(
+    arePackageDependenciesEqual(
+      {
+        version: '1.0',
+        dependencies: { a: '1' },
+      },
+      {
+        name: '1.1',
+        dependencies: { a: '1' },
+      }
+    )
+  ).toBe(true);
+});
+
+it('returns false if multiple dependency fields differ', () => {
+  expect(
+    arePackageDependenciesEqual(
+      {
+        dependencies: { a: '1' },
+        devDependencies: { c: '3' },
+      },
+      {
+        dependencies: { a: '1', b: '2' },
+        devDependencies: { c: '3', d: '4' },
+      }
+    )
+  ).toBe(false);
+});
+
+it('returns true if all dependency fields are the same', () => {
+  expect(
+    arePackageDependenciesEqual(
+      {
+        dependencies: { a: '1', b: '2', c: '3' },
+        devDependencies: { d: '4', e: '5', f: '6' },
+        peerDependencies: { g: '7', h: '8', i: '9' },
+      },
+      {
+        dependencies: { a: '1', b: '2', c: '3' },
+        devDependencies: { d: '4', e: '5', f: '6' },
+        peerDependencies: { g: '7', h: '8', i: '9' },
+      }
+    )
+  ).toBe(true);
+});
+
+it('returns true if non-dependency fields are added or removed', () => {
+  expect(
+    arePackageDependenciesEqual(
+      {
+        name: 'foo',
+        scripts: { test: 'yarn jest --watch' },
+        dependencies: { a: '1' },
+      },
+      {
+        name: 'foo',
+        license: 'MIT',
+        dependencies: { a: '1' },
+      }
+    )
+  ).toBe(true);
+});
+
+it('returns false if dependency objects are added', () => {
+  expect(arePackageDependenciesEqual({}, { dependencies: { a: '1' } })).toBe(false);
+});
+
+it('returns false if dependency objects are removed', () => {
+  expect(arePackageDependenciesEqual({ dependencies: { a: '1' } }, {})).toBe(false);
+});
```

Comment:
One more scenario: moving a dep from `dependencies` to `devDependencies`, all else staying equal.
We could allow this because no versions are changing (the same packages end up getting installed), but it might cause issues if devDeps explicitly aren't installed (e.g. in CI) when they build their Storybook, in which case there might actually be visual changes as a result. So probably safest to just bail in this scenario.

Reply:
Sounds good, shouldn't need a behavior change, I'll add a test!